packages/edit-site/src/components/navigation-sidebar/index.js

```diff
@@ -16,10 +17,25 @@ export const {
 } = createSlotFill( 'EditSiteNavigationPanelPreview' );
 
 export default function NavigationSidebar( {
-	defaultIsOpen = false,
+	isDefaultOpen = false,
 	activeTemplateType,
 } ) {
-	const [ isNavigationOpen, setIsNavigationOpen ] = useState( defaultIsOpen );
+	const isDesktopViewport = useViewportMatch( 'medium' );
+	const [ isNavigationOpen, setIsNavigationOpen ] = useState(
+		isDefaultOpen && isDesktopViewport
+	);
+
+	useEffect( () => {
+		// When transitioning to desktop open the navigation if `isDefaultOpen` is true.
+		if ( isDefaultOpen && isDesktopViewport && ! isNavigationOpen ) {
+			setIsNavigationOpen( true );
+		}
+
+		// When transitioning to mobile/tablet, close the navigation.
+		if ( ! isDesktopViewport && isNavigationOpen ) {
+			setIsNavigationOpen( false );
+		}
+	}, [ isDefaultOpen, isDesktopViewport, isNavigationOpen ] );
```

Comment:
I don't think we need to include `isNavigationOpen` here so that it won't accidentally triggers the effect. Also no need to conditionally check it before setting state, it will simply be a no-op if the states are identical.

Reply:
Good point 👍 

I've changed the code around a bit now so this code is a bit different anyway.

Reply:
Actually, ignore that, I thought I had it working, but it needs a few more changes to work on mobile.

Reply:
I've changed it back to your suggestion